etc/kayobe/kolla.yml

```
@@ -216,23 +300,27 @@ stackhpc_ubuntu_focal_repos:
 # Dict mapping Jinja2 block names in kolla's Docker images to their contents.
 kolla_build_blocks:
   base_header: |
-    {% if kolla_base_distro == 'centos' %}
+    {% if kolla_base_distro in ['centos', 'rocky'] %}
     RUN \
-    {% for repo in stackhpc_centos_stream_repos %}
-        sed -i -e 's/^\(mirrorlist *=.*\)/#\1/g' \
-               -e 's/^[# ]*\(baseurl *=.*\)/#\1/g' \
-               -e '/#baseurl.*/a baseurl={{ repo.url }}' /etc/yum.repos.d/{{ repo.file }}{% if not loop.last %} && \
+    {% for repo in stackhpc_yum_repos %}
+        sed -i -e '/\[{{ repo.tag }}\]/,/^\[/ s/^\(mirrorlist *=.*\)/#\1/g' \
+               -e '/\[{{ repo.tag }}\]/,/^\[/ s/^[# ]*\(baseurl *=.*\)/#\1/g' \
+               -e '/\[{{ repo.tag }}\]/,/^\[/ s/^[# ]*\(metalink *=.*\)/#\1/g' \
+               -e '/\[{{ repo.tag }}\]/,/^\[/ s|^\(name.*\)|\1 \nbaseurl={{ repo.url }}|' /etc/yum.repos.d/{{ repo.file }}{% if not loop.last %} && \
```

nit: remove space after name?
```suggestion
               -e '/\[{{ repo.tag }}\]/,/^\[/ s|^\(name.*\)|\1\nbaseurl={{ repo.url }}|' /etc/yum.repos.d/{{ repo.file }}{% if not loop.last %} && \
```

---

Done